### ISSUE TITLE
myth --init-db crashes with jsonrpc error.

### ISSUE THREAD

**Issue by flurbo2** [ConsenSys/mythril#113]
mythril crashes on every startup, when trying to resume from block 162

The following error is printed when running myth --init-db;

```
$ myth --init-db                                                                                                                                                                                                                        
Resuming synchronization from block 162
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/mythril/rpc/client.py", line 58, in _call
    return response['result']
KeyError: 'result'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/usr/lib/python3.6/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "/usr/local/lib/python3.6/dist-packages/mythril/ether/contractstorage.py", line 58, in sync_blocks
    block = self.eth.eth_getBlockByNumber(blockNum)
  File "/usr/local/lib/python3.6/dist-packages/mythril/rpc/base_client.py", line 75, in eth_getBlockByNumber
    return self._call('eth_getBlockByNumber', [block, tx_objects])
  File "/usr/local/lib/python3.6/dist-packages/mythril/rpc/client.py", line 60, in _call
    raise BadResponseError(response)
mythril.rpc.exceptions.BadResponseError: {'jsonrpc': '2.0', 'id': 1, 'error': {'code': -32602, 'message': 'invalid argument 0: hex string without 0x prefix'}}
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/bin/myth", line 253, in <module>
    contract_storage.initialize(eth)
  File "/usr/local/lib/python3.6/dist-packages/mythril/ether/contractstorage.py", line 110, in initialize
    results = pool.map(self.sync_blocks, numbers)
  File "/usr/lib/python3.6/multiprocessing/pool.py", line 266, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/usr/lib/python3.6/multiprocessing/pool.py", line 644, in get
    raise self._value
mythril.rpc.exceptions.BadResponseError: {'jsonrpc': '2.0', 'id': 1, 'error': {'code': -32602, 'message': 'invalid argument 0: hex string without 0x prefix'}}

```


The problem first appeared with a different error message;

```
[snip]
4038656 blocks processed, 11 unique contracts in database, next block: 22690
4040704 blocks processed, 11 unique contracts in database, next block: 20642
4042752 blocks processed, 11 unique contracts in database, next block: 18594
4044800 blocks processed, 11 unique contracts in database, next block: 16546
4046848 blocks processed, 11 unique contracts in database, next block: 14498
4048896 blocks processed, 11 unique contracts in database, next block: 12450
4050944 blocks processed, 11 unique contracts in database, next block: 10402
4052992 blocks processed, 11 unique contracts in database, next block: 8354
4055040 blocks processed, 11 unique contracts in database, next block: 6306
4057088 blocks processed, 11 unique contracts in database, next block: 4258
4059136 blocks processed, 11 unique contracts in database, next block: 2210
4061184 blocks processed, 11 unique contracts in database, next block: 162
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/mythril/rpc/client.py", line 58, in _call
    return response['result']
KeyError: 'result'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/usr/lib/python3.6/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "/usr/local/lib/python3.6/dist-packages/mythril/ether/contractstorage.py", line 59, in sync_blocks
    block = self.eth.eth_getBlockByNumber(blockNum)
  File "/usr/local/lib/python3.6/dist-packages/mythril/rpc/base_client.py", line 376, in eth_getBlockByNumber
    return self._call('eth_getBlockByNumber', [block, tx_objects])
  File "/usr/local/lib/python3.6/dist-packages/mythril/rpc/client.py", line 60, in _call
    raise BadResponseError(response)
mythril.rpc.exceptions.BadResponseError: {'jsonrpc': '2.0', 'id': 1, 'error': {'code': -32602, 'message': 'invalid argument 0: hex string without 0x prefix'}}
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/bin/myth", line 240, in <module>
    contract_storage.initialize(eth)
  File "/usr/local/lib/python3.6/dist-packages/mythril/ether/contractstorage.py", line 111, in initialize
    results = pool.map(self.sync_blocks, numbers)
  File "/usr/lib/python3.6/multiprocessing/pool.py", line 266, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/usr/lib/python3.6/multiprocessing/pool.py", line 644, in get
    raise self._value
mythril.rpc.exceptions.BadResponseError: {'jsonrpc': '2.0', 'id': 1, 'error': {'code': -32602, 'message': 'invalid argument 0: hex string without 0x prefix'}}
```

**Comment by b-mueller** [comment]
Hi @flurbo2,

I appears to me that it's syncing up to block 0 and then crashes for some reason - most likely an issue with the multithreading implementation. I'll look into it.

**Comment by gaasedelen** [comment]
This might be related to the threading work distribution. While I had a working DB, I noticed that scraping would eventually proceed into 'negative' block numbers upon reaching / nearing block zero.

eg `... unique contracts in database, next block: -112` and promptly crash.

**Comment by b-mueller** [comment]
Should be resolved in Mythril 0.16.1.
